### DISCUSSION TITLE
Resources or sample projects for CloudFlare workers?

### DISCUSSION THREAD

**Discussion by dikkadev** [leptos-rs/leptos#1234] [General]
The title really says it all. Are there any resources or (sample/real) projects on how to nicely integrate Leptos with CloudFlare workers? As I understand, it would mostly be about integration with provided Router, similar to actix and axum

**Comment by gbj** [comment]
Great question! There have been several efforts at Cloudflare Worker integrations. Nothing that's polished/public like the Actix/Axum/Viz integrations, but
- Here's an example using Cloudflare https://github.com/perry-birch/leptos-cloudflare 
- Here's an example using Deno, which shares some of the same JavaScript Fetch API Request/Response types https://github.com/seanaye/leptos-deno

I know @seanaye has been working on a Fetch integration so may have more to say here.

**Reply by seanaye** [reply]
The main reason I was working on the Fetch API integration is so that I could to deploy to an "edge" target E.g. cloudflare workers or Deno Deploy. 

Recently wasmedge has been announced which is a `wasi-unknown` edge deploy target. This would be strictly better than a fetch api integration for my use case, so I dropped working on the integration. 

It still appears like there is sufficient demand for it though so I will probably pick it back up

**Comment by dikkadev** [comment]
So I've started trying to copy/port the integration from axum/actix over to CloudFlare worker.
[cloudflare-worker-integration fork](https://github.com/Sett17/leptos/tree/cloudflare-worker-integration)

I *think* the server function things are more or less complete. Any pointers on how I'd go about testing this? Is there a kind of test-suite somewhere or do I have to make my own small test project?

**Reply by dikkadev** [reply]
After working on this a bit more, I'd definitely take some help/support for this. It is a _bit_ over my skill level currently. Although that makes it a nice, challenging project!
